src/v/cloud_storage/remote.h

```diff
@@ -518,6 +519,24 @@ class remote
     seastar::gate& gate() { return _gate; };
     ss::abort_source& as() { return _as; }
 
+    /////////////////////////////
```

Comment:
Why serialization and upload are mixed together?
Let's not do reuploads in this case. The simplest thing to do is to have some dedicated serialization primitives for jl segments in a different place. The actual upload is trivial.